caleb-lab/test/test.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+//require chai, buffer [node], and the filesystem
+const expect = require('chai').expect;
+const fs = require('fs');
+
+//describe the full module
+describe('fs module', function(){
+  //describe what we're trying to do with the things in the module
+  describe('#readFile', function(){
+    //describe what we expect with the test
+    it('should read a file and produce a raw buffer', done => {
+      fs.readFile(`${__dirname}/../data/text.txt`, function(err, data){
+        let bufferData = data;
+
+        //we expect this to be the data that gets operated on in the index.js, and for it to be a type of buffer. I intend to ask for more clarification on the "from". I particularly hate writing things that I do not understand.
+        expect(typeof bufferData).to.equal(typeof Buffer.from(''));
+
+        //need more clarification on done(), but too tired to do a google search right now
+        done();
```

Comment:
Passing in ``done`` basically signals to the Mocha environment that we are testing async code and to properly execute that code in the order we want it to. We invoke ``done()`` at the end of our expect statements to end the test or else the test environment will just hang and we'll get a timeout error.